EIPS/eip-custom_errors_for_erc_tokens.md

```diff
@@ -0,0 +1,393 @@
+  - Use `ERC1155InsufficientBalance` or `ERC1155InsufficientApproval` instead.
+
+#### `ERC1155InvalidReceiver(address receiver)`
+
+Indicates a failure with the token receiver.
+Used in transfers.
+
+- MUST be used for disallowed transfers to the zero address.
+- MUST be used for disallowed transfers to non-ERC1155TokenReceiver contracts or those that reject a transfer. (eg. returning an invalid response in `onERC1155Received`).
+- MUST NOT be used for approval operations.
+
+#### `ERC1155InsufficientApproval(address operator, uint256 id)`
+
+Indicates a failure with the `operator`'s approval in a transfer.
+Used in transfers.
+
+- MUST be used when operator `isApprovedForAll(owner, operator, id)` is false.
+
+#### `ERC1155InvalidApprover(address approver)`
+
+Indicates a failure with the `approver` of a token to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals from the zero address.
+- MUST NOT be used for transfer operations.
+
+#### `ERC1155InvalidOperator(address operator)`
+
+Indicates a failure with the `operator` to be approved.
+Used in approvals.
+
+- MUST be used for disallowed approvals to the zero address.
+- MUST be used for disallowed approvals to the owner itself.
+- MUST NOT be used for transfer operations.
+  - Use `ERC1155InsufficientApproval` instead.
+
+#### `ERC1155InvalidArrayLength(uint256 idsLength, uint256 valuesLength)`
+
+Indicates an array length mismatch between `_ids` and `_values` in a `safeBatchTransferFrom` operation.
+Used in batch transfers.
+
+- MUST be used only if `idsLength` is different to `valuesLength`
+
+### Parameter Glossary
+
+| Name        | Description                                                                         |
+| ----------- | ----------------------------------------------------------------------------------- |
+| `sender`    | The address of whose token(s) (is/are) transferred from.                            |
+| `balance`   | Current balance for the interacting account.                                        |
+| `needed`    | Amount minimum required to perform an action.                                       |
+| `receiver`  | Address owner of the token(s) transferred.                                          |
+| `spender`   | Address not owner of the token(s) transferred, but allowed to operate on (it/them). |
+| `allowance` | Amount of token(s) a `spender` is allowed to operate with.                          |
+| `approver`  | Owner of the token(s) being approved to an `spender`.                               |
+| `tokenId`   | The identifier number of a token type.                                              |
+| `owner`     | Address of the owner of a token type.                                               |
+| `operator`  | Same as `spender`.                                                                  |
+| `id`        | Same as `tokenId`.                                                                  |
```

Comment:
I agree I like it better if we consolidate, but is a similar case of using `Approval` vs `Allowance`. We're just keeping consistency with the standard-defined name.

Although I'd agree with consolidating, I'd like to ask why is this consolidatable and not `Approval`? Friendly question 😛

Reply:
Decided to consolidate for `tokenId` since the ERC1155 spec also mentions it, and parameter names are not considered as important as error names (for `Approval` vs `Allowance`).